src/main/java/uk/gov/ons/census/casesvc/service/NewAddressReportedService.java

```diff
@@ -0,0 +1,129 @@
+package uk.gov.ons.census.casesvc.service;
+
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.casesvc.logging.EventLogger;
+import uk.gov.ons.census.casesvc.model.dto.CollectionCase;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.entity.EventType;
+import uk.gov.ons.census.casesvc.utility.JsonHelper;
+
+@Component
+public class NewAddressReportedService {
+  private final CaseService caseService;
+  private final EventLogger eventLogger;
+
+  public NewAddressReportedService(CaseService caseService, EventLogger eventLogger) {
+    this.caseService = caseService;
+    this.eventLogger = eventLogger;
+  }
+
+  public void processNewAddress(
+      ResponseManagementEvent newAddressEvent, OffsetDateTime messageTimestamp) {
+    CollectionCase newCollectionCase =
+        newAddressEvent.getPayload().getNewAddress().getCollectionCase();
+    validDateNewAddressCollectionCaseForMandatoryFields(newCollectionCase);
+
+    Case skellingtonCase = createNewSkellingtonCase(newCollectionCase);
```

Comment:
Again, for the sake of anybody searching our codebase in future for "skeleton" this is really going to cause them to struggle to find code like this, so please rename.